Core/Scripts: Rewrite Boss Kalecgos (Sunwell Plateau)

sirikfoll opened the pull request:
<!--- (**********************************)
      (** Fill in the following fields **)
      (**********************************) --->

**Changes proposed:**

-  Scripted all spells
-  Corrected behavior of Spectral Blast and Curse of Boundless Agony
-  Removed wrong wipe behavior
-  Fix problem where bosses would be stuck with banish
-  Added Boundaries
-  Corrected texts
-  Removed old hacks
-  Other minor fixes

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Tests performed:** (Does it build, tested in-game, etc.) Built and tested, with 2 players

clopezramos commented on src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp:
```diff
@@ -16,523 +15,590 @@
+                case POINT_OUTRO_2:
+                    me->SetVisible(false);
+                    DespawnPortals();
+                    me->KillSelf();
```
despawnorunsummon

Keader commented on src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp:
```diff
@@ -572,239 +624,200 @@ class kalecgos_teleporter : public GameObjectScript
-                EnterEvadeMode();
-                return;
+                Unit* target = GetHitUnit();
+                target->CastSpell(target, SPELL_SPECTRAL_REALM_TELEPORT, true);
```
missing validator for these spells

Keader commented:
Click in Green button 👇

sirikfoll commented:
haha, I would like some users feedback first

Aokromes commented:
Conflicting files
src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp

sirikfoll committed:
Core/Scripts: Rewrite Boss Kalecgos (Sunwell Plateau)

Scripted all spells
Corrected behavior of Spectral Blast and Curse of Boundless Agony
Removed wrong wipe behavior
Fix problem where bosses would be stuck with banish
Added Boundaries
Corrected texts
Removed old hacks
Other minor fixes

sirikfoll committed:
rename

qxpf666 commented:
Conflicting files
src/server/scripts/EasternKingdoms/SunwellPlateau/boss_kalecgos.cpp

Aokromes commented:
this is pull request is part of trinitycore since 2 years ago.

sirikfoll merged commit e9cd7af into TrinityCore:3.3.5  on Aug 9, 2017

qxpf666 commented:
can someone update it?thanks

Keader commented:
Update? wtf?

Aokromes commented:
there is nothing to update, the code is part of trinitycore since 2 years ago, unless you are trying to use some ancient arcaic version we don't support.
